datafusion/functions/src/math/power.rs

```diff
@@ -92,8 +195,26 @@ impl ScalarUDFImpl for PowerFunc {
 
     fn return_type(&self, arg_types: &[DataType]) -> Result<DataType> {
```

Comment:
Can probably simplify to:

```rust
fn return_type(&self, arg_types: &[DataType]) -> Result<DataType> {
    Ok(arg_types[0].clone())
}
```

And assume the signature has done the checking for us

Reply:
Sorry, forgot it, done